### DISCUSSION TITLE
Collection contains only one equivalent item?

### DISCUSSION THREAD

**Discussion by tokarchyn** [fluentassertions/fluentassertions#2287] [Q&A]
Hi guys, does anyone know how to check that collection contains only **one** object that is equivalent to expected item? 
For example something like that:  `new []{new {a = "a"}, new {b = "b"}, new {b = "b"}}.Should().ContainSingleEquivalentTo(new {b = "b"})`, which should fail as there are two items matching the expected one.

**Comment by dennisdoomen** [comment]
There's no built-in API for this, but you could use `ContainSingle` with an expression that invokes `BeEquivalentTo` and catches the exception.
